CC-20877 Migration guide - Remove support of IE11 (#1204)
spryker/spryker-docs: docs/scos/dev/front-end-development/migration-guide-remove-support-of-ie11.md

---
title: Migration guide - Remove support of IE11
description: Use this guide to migrate the project from Yarn to NPM v8.
template: concept-topic-template
---

This document provides instructions for removing support of IE11 in your Spryker project.

## Overview

Microsoft dropped the support of IE11 in June 2022. So it has to be removed in Spryker.

*Estimated migration time: 1h*

## Update modules

Run the next command to update module versions:
```bash
composer update spryker/zed-ui spryker-shop/shop-ui
```

## Update configuration files

`.browserslistrc`:

Remove IE11 usage:
```text
IE 11
```

`frontend/configs/development.js`: 

1. Remove import of `es6-polyfill` file: 
```js

> 'es6-polyfill': es6PolyfillTs,

Reviewer: Missed about:

- Removing the `buildVariantSettings` variable declaration and usage.
- Removing the `es6PolyfillTs` variable.

Author: I thought it is clear that declarations and imports should be removed )

Reviewer: The migration can be done by different people and not everyone will analyze the code, but simply copy-paste it.
So it should be as detailed as possible.

```

2. Remove browsers support as it specified in the `.browserslistrc` file:
```js
browsers: [
    '> 1%',
    'ie >= 11',

> ],

Reviewer: Missed the `esmodules: true,`

```

3. Remove `isES6Module` usage from conditions, like: 
```js

> isES6Module ? `./js/${appSettings.name}.[name].js` : `./js/${appSettings.name}.[name].${buildVariant}.js`,

Reviewer: It should be replaced by 
``` 
filename: `./js/${appSettings.name}.[name].js`,
```
instead of removing

Author: ** Removing `isES6Module` usage from conditions ) Not the whole conditions

    
...

> ...(!isES6Module ? ['@babel/plugin-transform-runtime'] : []),

Reviewer: Also should be replaced by `['@babel/plugin-transform-runtime']` instead of removing

    
...

> ...(isES6Module ? getAssetsConfig(appSettings) : []),

Reviewer: ```suggestion
...getAssetsConfig(appSettings),
```

> ...(isES6Module ? getAssetsConfig(appSettings) : []),

Reviewer: Missed `autoprefixer` and `watchLifecycleEventNames` changes.

```

`frontend/libs/command-line-parser.js`:

Remove `build determined module version` message: 
```js
.option('-m, --module <module>', 'build determined module version', globalSettings.buildVariants)
```

`frontend/libs/compiler.js`:

Remove console command from the config section:
```js
const buildVariant = buildVariantSettings.buildVariant;

> console.log(`${config.namespace} (${config.theme}) building ${buildVariant} modules for ${config.webpack.mode}...`);

Reviewer: Missed the `const { buildVariantSettings } = require('../settings');` change.

Remove all IE11 related polyfills from the `vendor.ts`.

## Update templates

`src/Pyz/Yves/ShopUi/Theme/default/templates/page-blank/page-blank.twig`:

Remove `isNewFrontendBuildSupported` variable from the `template` block:
```twig
{%- raw -%}
{% block template %}
    {% set isNewFrontendBuildSupported = true %}

    ...
{% endblock %}
{% endraw %}
```

`src/Pyz/Zed/ZedUi/Presentation/Layout/layout.twig`:

Create `layout.twig` template with the next content:
```twig
{%- raw -%}
{% extends '@Spryker:ZedUi/Layout/layout.twig' %}

{% block template %}
    {% set importConfig = importConfig | merge({ isDifferentialMode: false }) %}

    {{ parent() }}
{% endblock %}
{% endraw %}
```

## Update package.json

`package.json`:

1. Update YVES build commands to: 
```json
"yves": "node ./frontend/build development",
"yves:watch": "node ./frontend/build development-watch",
"yves:production": "node ./frontend/build production",
```

2. Remove IE11 related dependencies:
```json
"classlist-polyfill": "~1.2.0",
"date-input-polyfill": "~2.14.0",
"element-closest": "~3.0.0",
"element-remove": "~1.0.4",
"intersection-observer": "~0.10.0",
"string.prototype.startswith": "~0.2.0",

> "whatwg-fetch": "~3.4.0",

Reviewer: Please mention them without versions, because they can be different.

```

3. Update `autoprefixer` dependency:
```json
"autoprefixer": "~9.8.8",
```

## Install and build

1. Install dependencies:
```bash
npm install 
```

> 2. Build the project:

Reviewer: Please add as a first option install/build via docker `docker/sdk cli npm i` and `docker/sdk up --assets`.
Left your option as a second option but add mention that it's local install/build

template: concept-topic-template
---

This document provides instructions for removing support of IE11 in your Spryker project.

## Overview

Microsoft dropped the support of IE11 in June 2022. So it has to be removed in Spryker.

*Estimated migration time: 1h*

## Update modules

Run the next command to update module versions:
```bash
composer update spryker/zed-ui spryker-shop/shop-ui
```

## Update configuration files

`.browserslistrc`:

Remove IE11 usage:
```text
IE 11
```

`frontend/configs/development.js`: 

1. Remove `buildVariantSettings` variable declaration and usage:
```js
const { buildVariantSettings } = require('../settings');
...
const { buildVariant, isES6Module } = buildVariantSettings;
```

2. Remove `es6PolyfillTs` variable declaration and usage:
```js
const es6PolyfillTs = await findAppEntryPoint(appSettings.find.shopUiEntryPoints, './es6-polyfill.ts');
...
'es6-polyfill': es6PolyfillTs,
```

3. Remove browsers support as it specified in the `.browserslistrc` file:
```js
browsers: [
    '> 1%',
    'ie >= 11',
],
```

```js
plugins: [
    autoprefixer({
        'browsers': ['> 1%', 'last 2 versions']
    })
]
// should be

> plugins: [require('autoprefixer')],

Reviewer: You didn't mention that the require at the top should be removed so they will have two `require('autoprefixer')`

---
title: Migration guide - Remove support of IE11
last_updated: Aug 31, 2022

> description: Use this guide to migrate the project from Yarn to NPM v8.

Reviewer 2: ```suggestion
description: This document shows how to remove support of IE11 in your Spryker project.
```

```

### Update `frontend/libs/command-line-parser.js`

Remove the `build determined module version` message: 
```js
.option('-m, --module <module>', 'build determined module version', globalSettings.buildVariants)
```

### Update `frontend/libs/compiler.js`

1. Remove import of the `buildVariantSettings` variable:
```js
const { buildVariantSettings } = require('../settings');
```

2. Remove the console command from the config section:
```js
const buildVariant = buildVariantSettings.buildVariant;

console.log(`${config.namespace} (${config.theme}) building ${buildVariant} modules for ${config.webpack.mode}...`);
```

##№ Update `frontend/settings.js`

1. Remove build variants:
```js
buildVariants: {
    esm: 'esm',
    legacy: 'legacy',
},
```

```js
const buildVariantArray = process.argv.filter(argv => argv.includes('module'));
const buildVariant = buildVariantArray.length ? buildVariantArray[0].replace('module:', '') : '';

const buildVariantSettings = {
    buildVariant,
    isES6Module: buildVariant === globalSettings.buildVariants.esm,
};
```

2. Remove `buildVariantSettings` variable export:
```js
module.exports = {
    ...,
    buildVariantSettings,
};
```

> `src/Pyz/Yves/ShopUi/Theme/default/vendor.ts`: 

Reviewer 2: ```suggestion
### Update `src/Pyz/Yves/ShopUi/Theme/default/vendor.ts`
```

Remove all IE11-related polyfills from `vendor.ts`.

{% info_block infoBox %}

The `es6-polyfill.ts` file was removed because all polyfills were specified in `vendor.ts`.

{% endinfo_block %}

## Update templates

> To update `src/Pyz/Yves/ShopUi/Theme/default/templates/page-blank/page-blank.twig`, remove the `isNewFrontendBuildSupported` variable from the `template` block:

Reviewer 2: ```suggestion
1. In `src/Pyz/Yves/ShopUi/Theme/default/templates/page-blank/page-blank.twig`, remove the `isNewFrontendBuildSupported` variable from the `template` block:
```

```twig
{%- raw -%}
{% block template %}
    {% set isNewFrontendBuildSupported = true %}

    ...
{% endblock %}
{% endraw %}
```

> To update `src/Pyz/Zed/ZedUi/Presentation/Layout/layout.twig`, create the `layout.twig` template with the following content:

Reviewer 2: ```suggestion
2. Create the `src/Pyz/Zed/ZedUi/Presentation/Layout/layout.twig` template with the following content:
```
